RadClass/DiffSpectra.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+
+
+class DiffSpectra:
+    '''
+    Estimates a minimum background spectrum for each
+    timestamp/spectrum and saves a difference spectrum
+    between the two.
+
+    Attributes:
+    stride: The number of timestamps prior to the current
+        spectrum for finding the minimum background
+    '''
+
+    def __init__(self, stride=10):
+        self.stride = stride
+
+    def run(self, data):
+        # index all windows of length self.diff_stride into a tmp array
+        windows = [data[i-self.stride:self.stride] for i in range(self.stride-1, data.shape[0])]
+        # filter out any windows without enough bckg samples
+        # i.e. beginning of spectra w/ idx < self.diff_stride
+        windows = [x for x in windows if x.shape[0] == self.stride]
```

Comment:
Why would any windows be too short? Can't you guarantee that they are all long enough (see above)

Reply:
You're right, with the correct indexing (good catch above), there should be no need to "filter" windows without the requisite number of spectra.